stix_shifter_utils/utils/param_validator.py

```diff
@@ -0,0 +1,149 @@
+    
+    try:
+        with open(config_json_path) as mapping_file:
+            expected_configs = json.load(mapping_file)
+    except Exception as ex:
+        raise(ex)
+    
+    validated_params = {}
+    errors = []
+    copy_valid_configs(input_configs, expected_configs, validated_params, errors)
+
+    error_obj = {}
+    if errors:
+        error_obj['missing_params'] = errors
+    if input_configs:
+        if isinstance(input_configs, dict):
+            error_obj['unexpected_params'] = get_inner_keys(input_configs)
+
+    if error_obj:
+        raise ValueError(error_obj)
+
+    return validated_params
+
+
+def copy_valid_configs(input_configs, expected_configs, validated_params, errors=[], current_path=''):
+    if isinstance(expected_configs, dict):
+        for key, value in expected_configs.items():
+            key_path = current_path
+            if key_path:
+                key_path = key_path + '.'
+            key_path = key_path + key
+            if key in input_configs:
+                if is_leaf(expected_configs[key]):
+                    if 'min' in expected_configs[key]:
+                        if not check_min(input_configs[key], expected_configs[key]['min']):
+                            raise ValueError('\"{}: {}\" value must be more than {}'.format(key, str(input_configs[key]), str(expected_configs[key]['min'])))
+                    elif 'max' in expected_configs[key]:
+                        if not check_max(input_configs[key], expected_configs[key]['max']):
+                            raise ValueError('\"{}: {}\" value must be less than {}'.format(key, str(input_configs[key]), str(expected_configs[key]['max'])))
+                    elif 'regex' in expected_configs[key]:
+                        if not check_regex(input_configs[key], expected_configs[key]['regex']):
+                            raise ValueError('Invalid {} value \"{}\" specified'.format(key, str(input_configs[key])))
+
+                    validated_params[key] = input_configs[key]
+                    del input_configs[key]
+                else:
+                    if key not in validated_params:
+                        validated_params[key] = dict()
+                    copy_valid_configs(input_configs[key], expected_configs[key], validated_params[key], errors, key_path)
+                    if not input_configs[key]:
+                        del input_configs[key]
+            else:
+                if not 'optional' in expected_configs[key] and not 'default' in expected_configs[key]:
+                    errors.append(key_path)
+                
+                if 'default' in expected_configs[key]:
+                    validated_params[key] = value['default']
+
+
+def is_leaf(config):
+    if isinstance(config, dict):
+        for key,value in config.items():
+            if isinstance(value, dict):
+                return False
+    return True
+
+def check_min(input_value, min_value):
+    if input_value >= min_value:
```

Comment:
it is wrong.
max check is wrong as well

Reply:
https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_modules/base_config.json#L7
min/max checking should work to check string length too.

Reply:
so the idea is, if `"type": "text"` check length if `"type": "number"` check value